continuous-deployment/readme.md

```diff
@@ -1,7 +1,142 @@
+
+* _Production sizing and capacity planning: How much data will your live application create? How many log files or databases will you need? How much bandwidth and disk space will you need? What latency are clients expecting?_
+
+* _How the initial deployment to production works._
+
+* _How fixing defects and applying patches to the production environment will be handled._
+
+* How upgrades to the production environment will be handled, including data migration. How will upgrades be carried out to the application without destroying its state.
+
+### Application Release and Environment Promotion
+
+Your release manifestation process should take the binaries created from your commit stage and deploy them across all cloud environments, starting with your test environment.
+
+The test environment (_often called Integration_) acts as a gate to validate your test suite completes successfully for all release candidates.
+
+#### The First Deployment
+
+The very first deployment of any application should be showcased to the customer in a production-like environment (_UAT_) to solicit feedback early and often.
+
+#### Criteria for a production-like environment
+
+* Runs the same operating system as production.
+* Has the same software installed as production.
+* Is sized and configured the same way as production.
+* Mirrors production's networking topology.
+* Simulated production-like load tests are executed following a release to surface any latency or throughput degradation.
+
+#### Modeling your Release Pipeline
+
+It's critical to model your test and release process to establish a common understanding between the application engineers and customer stakeholders. Specifically aligning expectations for how many cloud environments need to be pre-provisioned as well as defining sign-off gate roles and responsibilities.
+
+![image](https://user-images.githubusercontent.com/7635865/89290276-426ca200-d61e-11ea-9c79-bd183dc72050.png)
+
+##### The release pipeline model should depict:
+
+* All stages an application change would have to go through before it is released to production.
+* All release gate controls.
+* Customer-specific Cloud RBAC groups which have the authority to approve release candidates per environment.
+
+#### Release Pipeline Stages
+
+The stages within your release workflow are ultimately testing a version of your application to validate it can be released in accordance to your acceptance criteria. The release pipeline should account for the following conditions:
+
+* Release Selection: The developer carrying out application testing should have the capability to select which release version to deploy to the testing environment.
+* Environment Readiness - Clean the target cloud environment so it is ready to host a new version of the application.
+* Deployment - Release the application binary(s) to the target cloud environment.
```

Comment:
Again the word binary may cause someone to assume this only applies to compiled code. Perhaps rather than trying to land a more generic term, we can just clarify somewhere at the top what binary means in the context of this doc; which i think is basically any deployable build artifact.